pep-0683.rst

```diff
@@ -343,27 +469,15 @@ Constraints
 * be careful when immortalizing objects that we don't actually expect
   to persist until runtime finalization.
 * be careful when immortalizing objects that are not otherwise immutable
+* ``__del__`` and weakrefs must continue working properly
 
-.. _scope:
-
-Scope of Changes
-----------------
-
-Object immortality is not meant to be a public feature but rather an
-internal one.  So the proposal does *not* include adding any new
-public C-API, nor any Python API.  However, this does not prevent
-us from adding (publicly accessible) private API to do things
-like immortalize an object or tell if one is immortal.
-
-The particular details of:
-    
-* how to mark something as immortal
-* how to recognize something as immortal
-* which subset of functionally immortal objects are marked as immortal
-* which memory-management activities are skipped or modified for immortal objects
-    
-are not only Cpython-specific but are also private implementation
-details that are expected to change in subsequent versions.
+Regarding "truly" immutable objects, this PEP doesn't impact the
+effective immutability of any objects, other than the per-object
+runtime state (e.g. refcount).  So whether or not some immortal object
+is truly (or even effectively) immutable can only be settled separately
+from this proposal.  For example, str objects are generally considered
+immutable, but ``PyUnicodeObject`` holds some lazily cached data.  This
+PEP has no influence on how that state affects str immutability.
 
 Immortal Mutable Objects
```

Comment:
Nit: Since we already have a more detailed `Object Cleanup` it might be worth removing the piece that says:

```
(Note that we are still investigating the impact on GC
of immortalizing containers.)
```